cmd/agent/subcommands/flare/command.go

```diff
@@ -154,40 +157,60 @@ func readProfileData(seconds int) (flare.ProfileData, error) {
 
 	pdata := flare.ProfileData{}
 	c := util.GetClient(false)
+	sysProbeClient := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", pkgconfig.SystemProbe.GetString("system_probe_config.sysprobe_socket"))
+			},
+		},
+	}
+
+	type pprofGetter func(path string) ([]byte, error)
+
+	tcpGet := func(portConfig string) pprofGetter {
+		pprofURL := fmt.Sprintf("http://127.0.0.1:%d/debug/pprof", pkgconfig.Datadog.GetInt(portConfig))
+		return func(path string) ([]byte, error) {
+			return util.DoGet(c, pprofURL+path, util.LeaveConnectionOpen)
+		}
+	}
 
-	serviceProfileCollector := func(portConfig string, seconds int) agentProfileCollector {
+	sysProbeGet := func() pprofGetter {
+		return func(path string) ([]byte, error) {
+			return util.DoGet(sysProbeClient, "http://unix/debug/pprof"+path, util.LeaveConnectionOpen)
```

Comment:
I didn't write the original code, but my guess here would be that the flare command is going to exit soon anyway, so the connections should get closed.

I also updated this code to use the `RemoteSystemProbeUtil` object since that handles the windows case transparently. That object does not do closes either I believe since it is a singleton.